frb_example/benches/rust/src/executor.rs

```diff
@@ -0,0 +1,237 @@
+use std::{
+    panic::UnwindSafe,
+    sync::{Arc, Mutex},
+};
+
+use flutter_rust_bridge::{
+    handler::{
+        self, ErrorHandler, Executor, ReportDartErrorHandler, SimpleHandler, ThreadPoolExecutor,
+    },
+    rust2dart::TaskCallback,
+    IntoDart, MessagePort, WrapInfo,
+};
+
+use crate::api::{Metric, Unit};
+const ERROR_MUTEX_LOCK: &str = "Error on mutex lock";
+
+lazy_static::lazy_static! {
+  static ref METRICS: Arc<Mutex<Vec<Metric>>> = Arc::new(Mutex::new(vec![]));
+}
+
+pub type BenchHandler = SimpleHandler<BenchExecutor, BenchErrorHandler>;
+
+pub trait Metrics {
+    fn metrics(&self) -> Vec<Metric>;
+}
+
+impl Metrics for SimpleHandler<BenchExecutor, BenchErrorHandler> {
+    fn metrics(&self) -> Vec<Metric> {
+        let guard = METRICS.lock().expect(ERROR_MUTEX_LOCK);
+        guard.clone()
+    }
+}
+
+/// record a benchmark metric
+///
+/// all benchmark metrics are stored in [`METRICS`]
+fn record(debug_name_string: &str, elapsed: u64, unit: Unit) {
```

Comment:
nit: Maybe we can create a struct, say, `pub struct Metrics`,  and then all functions are methods on that struct, say, `impl Metrics {fn record(&self, ...) {...}}`. Then, we avoid many usages about singletons.

Reply:
Ah ok I'm gonna look into it.

Reply:
I can implement it on `BenchHandler`, but I'm not sure that's what you want here.
on the other side `METRICS` has to be a static ref otherwise it cannot outlive function calls.

Reply:
```rs
lazy_static::lazy_static! {
  static ref METRICS: Arc<Mutex<TheMetricsStruct>> = Arc::new(Mutex::new(TheMetricsStruct::new()));

struct TheMetricsStruct {
  metrics: Vec<Metric>,
}

impl TheMetricsStruct {
  fn record(&self, ) {...}
}
}
```

Reply:
I'm trying right now it's possible indeed to create this intermediate struct, but the trait definition (the one I called `Metrics` initially) cannot be removed because the `Handler` needs to implement it : in its original form the `Handler` doesn't provide public access to its fields, understandably. So it needs a trait so that `metrics()` function can be exposed and called from outside.

Reply:
trait LGTM, just saying the code above may cause as little as possible usage of *direct* access to global var and maybe better code style :)

Reply:
Ok I'll make the change ! 👌

Reply:
made the changes across 74d03d2, 4d828d6 and 4126097.

I leave this unresolved so you can confirm it's correct (or not) :)

Reply:
+ e82cb6d  (omitted change)